app/views/editapplication/credentialsPartials/fields.scala.html

```diff
@@ -51,10 +53,10 @@ <h2 class="heading-large" id="clientSecretHeading">Client secrets</h2>
         value = Some("OAuth 2.0")).toHtml journey. Keep them private.
     </p>
     @for((secret, i) <- token.clientSecrets.zipWithIndex) {
-        @maskedContent(secret, accessibleText = s"client secret ${i + 1}", secure = app.deployedTo == Environment.PRODUCTION, action = controllers.routes.Credentials.getProductionClientSecret(app.id, i).url)
+        @maskedContent(secret, accessibleText = s"client secret ${i + 1}", secure = app.deployedTo.isProduction, action = controllers.routes.Credentials.getProductionClientSecret(app.id, i).url)
     }
 
-    @if(app.canEditCredentials(developer)){
+    @if(app.allows(EditCredentials, developer, SandboxOrAdmin)){
```

Comment:
Should this SandboxOrAdmin be hidden away somewhere, so the view only knows about the capability and developer?

Reply:
For future consideration